src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -786,6 +786,46 @@ internal HttpRequestMessage BuildSetTriggersRequest()
             }
         }
 
+        public async Task<TriggersResult> GetTriggersAsync()
+        {
+            var result = new TriggersResult
+            {
+                Success = true
+            };
+
+            if (!IsSyncTriggersEnvironment(_webHostEnvironment, _environment))
+            {
+                result.Success = false;
+                result.Error = "Invalid environment for GetTriggers operation.";
+                _logger.LogWarning(result.Error);
+                return result;
+            }
+
+            try
+            {
+                await _syncSemaphore.WaitAsync();
+
+                PrepareSyncTriggers();
```

Comment:
I think we should move PrepareSyncTriggers into GetSyncTriggersPayload, since it needs to be done in all cases, and we ensure that it's always done any time GetSyncTriggers is called. Wasn't an issue when we had only one caller, but now we have multiple

Reply:
Yes, moved PrepareSyncTriggers() call into GetSyncTriggersPayload()